tests/test_dia_calculation_plugins.py

```diff
@@ -53,39 +56,149 @@ def testCalculate(self):
                                "ap_meanPosition",
                                None)
         plug.calculate(diaObject, diaSources)
 
         self.assertAlmostEqual(diaObject["ra"], 0.0)
         self.assertAlmostEqual(diaObject["decl"], 0.0)
+        self.assertEqual(diaObject["radecTai"], 10)
 
         diaObject = dict()
         diaSources = pd.DataFrame(data={"ra": np.zeros(n_sources),
-                                        "decl": np.linspace(-1, 1, n_sources)})
+                                        "decl": np.linspace(-1, 1, n_sources),
+                                        "midPointTai": np.linspace(0,
+                                                                   n_sources,
+                                                                   n_sources)})
         plug.calculate(diaObject, diaSources)
 
         self.assertAlmostEqual(diaObject["ra"], 0.0)
         self.assertAlmostEqual(diaObject["decl"], 0.0)
+        self.assertEqual(diaObject["radecTai"], 10)
 
         # Test failure modes.
         diaObject = dict()
         diaSources = pd.DataFrame(data={"ra": np.full(n_sources, np.nan),
-                                        "decl": np.zeros(n_sources)})
+                                        "decl": np.zeros(n_sources),
+                                        "midPointTai": np.linspace(0,
+                                                                   n_sources,
+                                                                   n_sources)})
         plug.calculate(diaObject, diaSources)
 
         self.assertTrue(np.isnan(diaObject["ra"]))
         self.assertTrue(np.isnan(diaObject["decl"]))
+        self.assertTrue(np.isnan(diaObject["radecTai"]))
 
         diaObject = dict()
         diaSources = pd.DataFrame(data={"ra": np.zeros(n_sources),
-                                        "decl": np.full(n_sources, np.nan)})
+                                        "decl": np.full(n_sources, np.nan),
+                                        "midPointTai": np.linspace(0,
+                                                                   n_sources,
+                                                                   n_sources)})
         plug.calculate(diaObject, diaSources)
 
         self.assertTrue(np.isnan(diaObject["ra"]))
         self.assertTrue(np.isnan(diaObject["decl"]))
+        self.assertTrue(np.isnan(diaObject["radecTai"]))
+
+
+class TestHTMIndexPosition(unittest.TestCase):
+
+    def testCalculate(self):
+        """Test HTMPixel assignment calculation.
+        """
+        # Test expected pixelIds.
+        diaObject = dict()
+        diaObject["ra"] = 0.
+        diaObject["decl"] = 0.
+        plug = HTMIndexDiaPosition(HTMIndexDiaPositionConfig(),
+                                   "ap_HTMIndex",
+                                   None)
+        plug.calculate(diaObject)
+        self.assertAlmostEqual(diaObject["pixelId"], 131072)
```

Comment:
Can you provide any context about why this (and the subsequent example) is the correct pixelId value?

Reply:
These are the values produced by the HTM pixelization at the locations requested. The specific values here aren't particularly important, only that they are stable and round trip correctly.